tracing/http_test.go

```diff
@@ -166,3 +166,49 @@ func TestHTTPSpanIgnoringPaths(t *testing.T) {
 		})
 	}
 }
+
+func BenchmarkSpanMiddleware(b *testing.B) {
+	cfg := config.Config{
+		Backend: config.Zipkin,
+		Debug:   true,
+	}
+
+	// Create tracer with reporter recorder
+	reporter, co := FakeZipkinExporter()
+	oct := NewOpenCensusTracer(co)
+	b.Cleanup(func() {
+		reporter.Close()
+		oct.Finish()
+	})
+
+	if err := oct.ApplyConfig(&cfg); err != nil {
+		b.Error("Failed to apply tracer config:", err)
+	}
+
+	next := testHandler{}
+	middleware := HTTPSpanMiddleware(&next)
+
+	fw := &fakeWriter{}
```

Comment:
Is anything ever calling `Write` on this? That function would cause additional allocations I believe if called.

Reply:
It is the baseline because the opencensus handler will not live alone but part of an http handler chain anyway. It's overhead seems minimal from the call graph:
![image](https://user-images.githubusercontent.com/7945591/113307118-77e0a100-930d-11eb-8126-4a3260baaa31.png)

Reply:
I can create a dummy doing nothing.

Reply:
Yeah if the tracing handler doesn't call Write (which I guess it shouldn't anyway), we should be fine.

Reply:
Actually its one allocation less (pretty predictible):
```
$ go test   -benchmem -memprofile memprofile.out -cpuprofile profile.out -test.v -test.bench BenchmarkSpanMiddleware -test.run ^$ ./tracing/
goos: linux
goarch: amd64
pkg: knative.dev/pkg/tracing
BenchmarkSpanMiddleware
BenchmarkSpanMiddleware/sequential
BenchmarkSpanMiddleware/sequential-12         	  141504	     10679 ns/op	    6911 B/op	      97 allocs/op
BenchmarkSpanMiddleware/parallel
BenchmarkSpanMiddleware/parallel-12           	  661308	      4737 ns/op	    6742 B/op	      97 allocs/op
PASS
ok  	knative.dev/pkg/tracing	5.204s
```

Reply:
Okay, no biggy then. Fine to keep as-is.

Reply:
It seems it calls the handler [here](https://github.com/census-instrumentation/opencensus-go/blob/de37041b23a4ac17ac695473ec3876c40a4b65f8/plugin/ochttp/server.go#L92).